app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision/cloning.rb

```diff
@@ -4,57 +4,71 @@ def log_clone_options(clone_options)
   end
 
   def prepare_for_clone_task
-    specs = {
-      'image_id'   => get_option_last(:src_vm_id),
-      'pin_policy' => get_option_last(:pin_policy),
-    }
+    if request_type == 'clone_to_template'
```

Comment:
@miyamotoh wonder if we'd be better off splitting these out into two methods rather than having 90% of the code under a conditional?
Assuming we validate the request_type is clone_to_template or clone_to_vm we could even `send("prepare_for_#{request_type}_task")`

Reply:
Agreed, @agrare . I'll try to make it that way.

Reply:
@agrare please take a look at my attempt in the latest push. Basically I tried to contain what you suggest in my provider code, as I wasn't sure if I wanted to change it in the core/parent, without worrying about all the providers I don't touch. The git diff looks a bit big, but basically I'm moving the logic to `private` funcs at the bottom of the file. Give me a guidance for a better implementation.